synapse/handlers/search.py

```diff
@@ -139,6 +187,27 @@ def search(self, user, content, batch=None):
 
         room_ids = search_filter.filter_rooms(room_ids)
 
+        # If doing a subset of all rooms seearch, check if any of the rooms
+        # are from an upgraded room, and search their contents as well
+        # XXX: There is the possibility that we don't have a create event for
```

Comment:
we ought to have the create event for any room that the user has ever been a member of, so I don't think this is relevant